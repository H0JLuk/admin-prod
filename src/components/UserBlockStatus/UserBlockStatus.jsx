import React from 'react';
import cn from 'classnames';

import styles from './UserBlockStatus.module.css';

const STATUS_BLOCKED = 'Заблокирован';
const STATUS_ACTIVE = 'Активен';

const UserBlockStatus = ({ blocked, withText, className = '' }) => (
    <div className={ className }>
        {withText && (
            blocked ? STATUS_BLOCKED : STATUS_ACTIVE
        ) }
        <div
            className={ cn({
                [styles.statusOff]: blocked,
                [styles.statusOn]: !blocked,
            }) }
        />
    </div>
);

export default UserBlockStatus;
