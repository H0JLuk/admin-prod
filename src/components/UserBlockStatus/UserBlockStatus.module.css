.statusOff,
.statusOn {
    width: 10px;
    height: 10px;
    border-radius: 50%;
    display: inline-block;
    margin-left: 5px;
}

.statusOff {
    background-color: #ff5867;
}

.statusOn {
    background-color: #86ee82;
}
