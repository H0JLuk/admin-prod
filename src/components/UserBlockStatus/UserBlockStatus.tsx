import React from 'react';
import cn from 'classnames';

import styles from './UserBlockStatus.module.css';

const STATUS_BLOCKED = 'Заблокирован';
const STATUS_ACTIVE = 'Активен';

export type UserBlockStatusProps = {
    blocked?: boolean;
    withText?: boolean;
    className?: string;
};

const UserBlockStatus: React.FC<UserBlockStatusProps> = ({ blocked, withText, className = '' }) => (
    <div className={className}>
        {withText && (
            blocked ? STATUS_BLOCKED : STATUS_ACTIVE
        )}
        <div
            className={cn({
                [styles.statusOff]: blocked,
                [styles.statusOn]: !blocked,
            })}
        />
    </div>
);

export default UserBlockStatus;
