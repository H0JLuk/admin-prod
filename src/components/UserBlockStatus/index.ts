import UserBlockStatus from './UserBlockStatus';

export type { UserBlockStatusProps } from './UserBlockStatus';

export default UserBlockStatus;
