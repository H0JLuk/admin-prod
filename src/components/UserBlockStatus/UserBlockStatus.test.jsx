import React from 'react';
import { shallow } from 'enzyme';
import UserBlockStatus from './UserBlockStatus';

const STATUS_BLOCKED = 'Заблокирован';
const STATUS_ACTIVE = 'Активен';

const TEST_CLASS = 'TEST_CLASS';

const TEST_PROPS_FALSE = {
    blocked: false,
    withText: false,
    className: TEST_CLASS,
};

const TEST_PROPS_TRUE = {
    blocked: true,
    withText: true,
    className: TEST_CLASS,
};

describe('<UserBlockStatus /> test', () => {
    const Component = shallow(<UserBlockStatus { ...TEST_PROPS_FALSE } />);

    it('should be mount', () => {
        expect(Component.html()).toMatchSnapshot();
    });

    it('should be class name status on', () => {
        const statusBlock = Component.find('.statusOn');
        expect(statusBlock.debug()).not.toBe('');
    });

    it('should be class name status off and title status blocked', () => {
        const Component = shallow(<UserBlockStatus { ...TEST_PROPS_TRUE } />);
        const statusBlock = Component.find('.statusOff');
        const statusTitle = Component.find(`.${TEST_CLASS}`);
        expect(statusBlock.debug()).not.toBe('');
        expect(statusTitle.text()).toBe(STATUS_BLOCKED);
    });

    it('should be class name status on and title status active', () => {
        const Component = shallow(<UserBlockStatus { ...TEST_PROPS_TRUE } blocked={ false } />);
        const statusBlock = Component.find('.statusOn');
        const statusTitle = Component.find(`.${TEST_CLASS}`);
        expect(statusBlock.debug()).not.toBe('');
        expect(statusTitle.text()).toBe(STATUS_ACTIVE);
    });

});
