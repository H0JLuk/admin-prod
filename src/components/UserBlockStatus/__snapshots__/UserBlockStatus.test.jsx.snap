// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBlockStatus /> test should be mount 1`] = `"<div class=\\"TEST_CLASS\\"><div class=\\"statusOn\\"></div></div>"`;
